### ISSUE TITLE
Question about the role of world-model outputs during inference

### ISSUE THREAD

**Issue by 2002-Pan** [Feliciaxyao/NavMorph#24]
Hi authors,

Thank you for the great work and for releasing the code. I have been reading the implementation of NavMorph and I am trying to better understand how the world-model branch is used during inference.

From my current understanding, the prediction branch takes the concatenated panorama embedding and position embedding as input:

    combined_embeds = torch.cat((vis_embeds, pos_embedding), dim=-1)
    wm_outputs, real_state, mu_posterior, sigma_posterior = self.policy.net(
        mode="prediction",
        pred_pano_embeds=combined_embeds,
    )

It seems that `wm_outputs` is used to predict the next position / motion delta and is supervised during training. The posterior-prior matching and observation prediction also seem to provide auxiliary world-model objectives.

During evaluation, I can see that CEM directly enhances the current panorama representation through `enhanced_pano_embeds`, which is then used in the graph-based navigation module. However, I am not fully sure how `wm_outputs` or the imagined trajectory directly affects the GraphMap or final `nav_logits` during inference.

I also noticed a `posref_update()` function that seems related to refining navigation scores using the predicted position, but I am not sure whether it is enabled in the released inference pipeline.

Could you please clarify this part? In the current implementation, should the world-model prediction branch be understood mainly as an auxiliary training signal for learning better dynamics-aware representations, while CEM is the component that directly updates the representation used by the navigation graph? Or is there another inference path where the world-model outputs directly refine the navigation decision?

Thanks again for your help!
